app/WebParsing/UtsgJsonParser.hs

```diff
@@ -91,14 +85,27 @@ meetingUpdates m = [ MeetingCode =. meetingCode m
                    , MeetingExtra =. meetingExtra m
                    ]
 
-newtype DB = DB { dbData :: (Courses, [MeetTime]) }
+newtype DB = DB { dbData :: [MeetTime]}
   deriving Show
 
 instance FromJSON DB where
-    parseJSON (Object o) = do
-      course <- parseJSON (Object o)
-      session :: T.Text <- o .:? "section" .!= "F"
-      meetingTimesMap :: HM.HashMap T.Text MeetTime <- o .:? "meetings" .!= HM.empty
-      let allMeetingsTimes = map (\m -> m {meetInfo = (meetInfo m) { meetingCode = coursesCode course, meetingSession = session}}) (HM.elems meetingTimesMap)
-      return $ DB (course, allMeetingsTimes)
-    parseJSON _ = fail "Invalid section"
+  parseJSON = withObject "Expected an Object for DB" $ \o -> do
+    course <- parseJSON (Object o)
+    session :: T.Text <- o .:? "sectionCode" .!= "F"
```

Comment:
Yikes, even though this was the previous default I don't think this is good. We shouldn't "guess" a section code. If there isn't one, we should let the parser fail (i.e. by using `o .: "sectionCode"` instead of `.:?`)